### DISCUSSION TITLE
How system selects leader for aggregation?

### DISCUSSION THREAD

**Discussion by srxw** [HewlettPackard/swarm-learning#138] [Q&A]
Hello, I have read all your documents, which mentioned that Swarm Learning will temporary select a SL node as a leader to aggregate local models, but it does not mention the selection rules of the leader. I want to know that more detail to customize my training.

**Comment by iArpanPatel** [comment]
@srxw apologies for late response.  

The first SL node to complete the local Training for a given sync round, will be elected as a leader. Please note the leader can change in each sync round.

There is a new Environment Variable ('SL_MAKE_ME_ADMIN') introduced in recent release, where user can make specific SL to opt out of the Leader Election process. Refer [here](https://github.com/HewlettPackard/swarm-learning/blob/master/docs/Install/Environment_variables.md) for more details.
